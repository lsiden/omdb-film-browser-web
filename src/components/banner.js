import React from "react"
import QueryForm from "query-form"
import { OMDB_URL } from "constants"

const headerStyle = {
  background: "linear-gradient(to right, #000040, #3b00ff)",
  padding: "10px",
  color: "white",
}
const titleStyle = {
  fontSize: "18pt",
  marginBottom: "18pt",
}

const linkStyle = {
  color: "white",
  fontSize: "small",
}

const Banner = () => (
  <header style={headerStyle}>
    <h1 style={titleStyle}>
      {"Search Open Movie Database"}&nbsp;
      <a href={OMDB_URL} style={linkStyle}>
        (OMDB)
      </a>
    </h1>
    <QueryForm />
  </header>
)

export default Banner
