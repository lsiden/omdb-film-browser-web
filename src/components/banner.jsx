import React from 'react'

import { headerStyle, titleStyle } from 'style'
import { BANNER_TITLE } from 'omdb_constants'

const AboutLink = () => (
  <a href="/about" style={{ color: 'white' }}>
    about
  </a>
)

const inAboutPath = () => {
  return /about\/?$/.test(window.location.href)
}

export default () => {
  return (
    <header style={headerStyle}>
      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <h1 style={titleStyle}>{BANNER_TITLE}</h1>
        {!inAboutPath() && <AboutLink />}
      </div>
    </header>
  )
}