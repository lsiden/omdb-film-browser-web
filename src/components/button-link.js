import React from "react"
import PropTypes from "prop-types"

const ButtonLink = (props) => <button className="btn btn-link" {...props} />

ButtonLink.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string])
    .isRequired,
}

export default ButtonLink
