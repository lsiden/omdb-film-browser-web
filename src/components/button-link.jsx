import React from 'react'
import PropTypes from 'prop-types'

const ButtonLink = ({ style, onClick, children }) => (
  /* eslint-disable-next-line jsx-a11y/control-has-associated-label */
  <button
    type="button"
    className="btn btn-link"
    style={style}
  >
    {children}
  </button>
)

ButtonLink.propTypes = {
  style: PropTypes.object,
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
}

ButtonLink.defaultProps = {
  style: {},
}

export default ButtonLink
