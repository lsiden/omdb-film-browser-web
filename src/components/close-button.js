import React from "react"
import { Link } from "react-router-dom"

import ButtonLink from "components/button-link"

const CLOSE_ICON = "\u2715"

const closeButtonStyle = {
  margin: "8pt",
  fontSize: "20pt",
}

export const CloseButton = () => (
  <Link to="/">
    <ButtonLink style={closeButtonStyle} title="Close">
      {CLOSE_ICON}
    </ButtonLink>
  </Link>
)

export default CloseButton
