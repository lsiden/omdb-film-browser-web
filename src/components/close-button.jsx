import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import ButtonLink from 'components/button-link'

const CLOSE_ICON = '\u2715'

const closeButtonStyle = {
  margin: 20,
  fontSize: 20,
}

const CloseButton = ({ buttonStyle, onClick }) => {
  const style = {
    ...closeButtonStyle,
    ...buttonStyle,
  }
  return (
    <Link to="/">
      <ButtonLink onClick={onClick} style={style} title="Close">
        {CLOSE_ICON}
      </ButtonLink>
    </Link>
  )
}
CloseButton.propTypes = {
  buttonStyle: PropTypes.object,
}
CloseButton.defaultProps = {
  buttonStyle: {},
}

export default CloseButton
