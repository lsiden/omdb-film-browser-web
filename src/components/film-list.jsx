import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import FilmTitle from 'components/film-title'
import { getFilms } from 'store'

const ulStyle = {
  listStyleType: 'none',
  lineHeight: 1.5,
  marginTop: 20,
}

export const _FilmList = ({ films }) => (
  <ul style={ulStyle}>
    {films.map((film) => (
      <FilmTitle key={film.imdbID} filmSummary={film} />
    ))}
  </ul>
)

_FilmList.propTypes = {
  films: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default connect(
  (state) => ({
    films: getFilms(state),
  }),
)(_FilmList)
