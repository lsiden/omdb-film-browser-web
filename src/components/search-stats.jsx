import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { getFilms } from 'store'

export const _SearchStats = ({ numFetched, numFound, style }) => (
  <div style={style}>
    Fetched {numFetched} of {numFound}
  </div>
)

_SearchStats.propTypes = {
  numFetched: PropTypes.number.isRequired,
  numFound: PropTypes.number.isRequired,
  style: PropTypes.object,
}

_SearchStats.defaultProps = {
  style: {},
}

export default connect(
  (state, ownProps) => ({
    numFetched: getFilms(state).length,
    numFound: state.totalResults || 0,
    style: ownProps.style,
  })
)(_SearchStats)
