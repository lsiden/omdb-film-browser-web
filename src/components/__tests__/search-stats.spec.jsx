import React from 'react'
import { create } from 'react-test-renderer'

import { _SearchStats } from 'components/search-stats'

const createWrapper = (numFetched, numFound) => create(
  <_SearchStats numFetched={numFetched} numFound={numFound} />
).root

describe('SearchStats', () => {
  it('matches snapshot', () => {
    expect(createWrapper(5, 10)).toMatchSnapshot()
  })
})
