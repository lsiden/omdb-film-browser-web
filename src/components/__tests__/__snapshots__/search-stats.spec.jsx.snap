// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchStats matches snapshot 1`] = `
ReactTestInstance {
  "_fiber": FiberNode {
    "_debugHookTypes": null,
    "_debugID": 3,
    "_debugIsCurrentlyTiming": false,
    "_debugNeedsRemount": false,
    "_debugOwner": null,
    "_debugSource": Object {
      "columnNumber": 3,
      "fileName": "/Users/lawrencesiden/projects/omdb-film-browser-web/src/components/__tests__/search-stats.spec.jsx",
      "lineNumber": 7,
    },
    "actualDuration": 0,
    "actualStartTime": -1,
    "alternate": null,
    "child": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 4,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "columnNumber": 3,
        "fileName": "/Users/lawrencesiden/projects/omdb-film-browser-web/src/components/search-stats.jsx",
        "lineNumber": 8,
      },
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 5,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": null,
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": "Fetched ",
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": "Fetched ",
        "ref": null,
        "return": [Circular],
        "selfBaseDuration": 0,
        "sibling": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 6,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": null,
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 1,
          "key": null,
          "lastEffect": null,
          "memoizedProps": "5",
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": "5",
          "ref": null,
          "return": [Circular],
          "selfBaseDuration": 0,
          "sibling": FiberNode {
            "_debugHookTypes": null,
            "_debugID": 7,
            "_debugIsCurrentlyTiming": false,
            "_debugNeedsRemount": false,
            "_debugOwner": null,
            "_debugSource": null,
            "actualDuration": 0,
            "actualStartTime": -1,
            "alternate": null,
            "child": null,
            "childExpirationTime": 0,
            "dependencies": null,
            "effectTag": 0,
            "elementType": null,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 2,
            "key": null,
            "lastEffect": null,
            "memoizedProps": " of ",
            "memoizedState": null,
            "mode": 0,
            "nextEffect": null,
            "pendingProps": " of ",
            "ref": null,
            "return": [Circular],
            "selfBaseDuration": 0,
            "sibling": FiberNode {
              "_debugHookTypes": null,
              "_debugID": 8,
              "_debugIsCurrentlyTiming": false,
              "_debugNeedsRemount": false,
              "_debugOwner": null,
              "_debugSource": null,
              "actualDuration": 0,
              "actualStartTime": -1,
              "alternate": null,
              "child": null,
              "childExpirationTime": 0,
              "dependencies": null,
              "effectTag": 0,
              "elementType": null,
              "expirationTime": 0,
              "firstEffect": null,
              "index": 3,
              "key": null,
              "lastEffect": null,
              "memoizedProps": "10",
              "memoizedState": null,
              "mode": 0,
              "nextEffect": null,
              "pendingProps": "10",
              "ref": null,
              "return": [Circular],
              "selfBaseDuration": 0,
              "sibling": null,
              "stateNode": Object {
                "isHidden": false,
                "tag": "TEXT",
                "text": "10",
              },
              "tag": 6,
              "treeBaseDuration": 0,
              "type": null,
              "updateQueue": null,
            },
            "stateNode": Object {
              "isHidden": false,
              "tag": "TEXT",
              "text": " of ",
            },
            "tag": 6,
            "treeBaseDuration": 0,
            "type": null,
            "updateQueue": null,
          },
          "stateNode": Object {
            "isHidden": false,
            "tag": "TEXT",
            "text": "5",
          },
          "tag": 6,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": null,
        },
        "stateNode": Object {
          "isHidden": false,
          "tag": "TEXT",
          "text": "Fetched ",
        },
        "tag": 6,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": null,
      },
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 0,
      "elementType": "div",
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": Array [
          "Fetched ",
          5,
          " of ",
          10,
        ],
        "style": Object {},
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": Array [
          "Fetched ",
          5,
          " of ",
          10,
        ],
        "style": Object {},
      },
      "ref": null,
      "return": [Circular],
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": Object {
        "children": Array [
          Object {
            "isHidden": false,
            "tag": "TEXT",
            "text": "Fetched ",
          },
          Object {
            "isHidden": false,
            "tag": "TEXT",
            "text": "5",
          },
          Object {
            "isHidden": false,
            "tag": "TEXT",
            "text": " of ",
          },
          Object {
            "isHidden": false,
            "tag": "TEXT",
            "text": "10",
          },
        ],
        "internalInstanceHandle": [Circular],
        "isHidden": false,
        "props": Object {
          "children": Array [
            "Fetched ",
            5,
            " of ",
            10,
          ],
          "style": Object {},
        },
        "rootContainerInstance": Object {
          "children": Array [
            [Circular],
          ],
          "createNodeMock": [Function],
          "tag": "CONTAINER",
        },
        "tag": "INSTANCE",
        "type": "div",
      },
      "tag": 5,
      "treeBaseDuration": 0,
      "type": "div",
      "updateQueue": null,
    },
    "childExpirationTime": 0,
    "dependencies": null,
    "effectTag": 1,
    "elementType": [Function],
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "numFetched": 5,
      "numFound": 10,
      "style": Object {},
    },
    "memoizedState": null,
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {
      "numFetched": 5,
      "numFound": 10,
      "style": Object {},
    },
    "ref": null,
    "return": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 1,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 1,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": [Circular],
        "child": null,
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 1073741823,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": FiberRootNode {
          "callbackExpirationTime": 0,
          "callbackNode": null,
          "callbackPriority": 90,
          "containerInfo": Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "isHidden": false,
                    "tag": "TEXT",
                    "text": "Fetched ",
                  },
                  Object {
                    "isHidden": false,
                    "tag": "TEXT",
                    "text": "5",
                  },
                  Object {
                    "isHidden": false,
                    "tag": "TEXT",
                    "text": " of ",
                  },
                  Object {
                    "isHidden": false,
                    "tag": "TEXT",
                    "text": "10",
                  },
                ],
                "internalInstanceHandle": FiberNode {
                  "_debugHookTypes": null,
                  "_debugID": 4,
                  "_debugIsCurrentlyTiming": false,
                  "_debugNeedsRemount": false,
                  "_debugOwner": [Circular],
                  "_debugSource": Object {
                    "columnNumber": 3,
                    "fileName": "/Users/lawrencesiden/projects/omdb-film-browser-web/src/components/search-stats.jsx",
                    "lineNumber": 8,
                  },
                  "actualDuration": 0,
                  "actualStartTime": -1,
                  "alternate": null,
                  "child": FiberNode {
                    "_debugHookTypes": null,
                    "_debugID": 5,
                    "_debugIsCurrentlyTiming": false,
                    "_debugNeedsRemount": false,
                    "_debugOwner": null,
                    "_debugSource": null,
                    "actualDuration": 0,
                    "actualStartTime": -1,
                    "alternate": null,
                    "child": null,
                    "childExpirationTime": 0,
                    "dependencies": null,
                    "effectTag": 0,
                    "elementType": null,
                    "expirationTime": 0,
                    "firstEffect": null,
                    "index": 0,
                    "key": null,
                    "lastEffect": null,
                    "memoizedProps": "Fetched ",
                    "memoizedState": null,
                    "mode": 0,
                    "nextEffect": null,
                    "pendingProps": "Fetched ",
                    "ref": null,
                    "return": [Circular],
                    "selfBaseDuration": 0,
                    "sibling": FiberNode {
                      "_debugHookTypes": null,
                      "_debugID": 6,
                      "_debugIsCurrentlyTiming": false,
                      "_debugNeedsRemount": false,
                      "_debugOwner": null,
                      "_debugSource": null,
                      "actualDuration": 0,
                      "actualStartTime": -1,
                      "alternate": null,
                      "child": null,
                      "childExpirationTime": 0,
                      "dependencies": null,
                      "effectTag": 0,
                      "elementType": null,
                      "expirationTime": 0,
                      "firstEffect": null,
                      "index": 1,
                      "key": null,
                      "lastEffect": null,
                      "memoizedProps": "5",
                      "memoizedState": null,
                      "mode": 0,
                      "nextEffect": null,
                      "pendingProps": "5",
                      "ref": null,
                      "return": [Circular],
                      "selfBaseDuration": 0,
                      "sibling": FiberNode {
                        "_debugHookTypes": null,
                        "_debugID": 7,
                        "_debugIsCurrentlyTiming": false,
                        "_debugNeedsRemount": false,
                        "_debugOwner": null,
                        "_debugSource": null,
                        "actualDuration": 0,
                        "actualStartTime": -1,
                        "alternate": null,
                        "child": null,
                        "childExpirationTime": 0,
                        "dependencies": null,
                        "effectTag": 0,
                        "elementType": null,
                        "expirationTime": 0,
                        "firstEffect": null,
                        "index": 2,
                        "key": null,
                        "lastEffect": null,
                        "memoizedProps": " of ",
                        "memoizedState": null,
                        "mode": 0,
                        "nextEffect": null,
                        "pendingProps": " of ",
                        "ref": null,
                        "return": [Circular],
                        "selfBaseDuration": 0,
                        "sibling": FiberNode {
                          "_debugHookTypes": null,
                          "_debugID": 8,
                          "_debugIsCurrentlyTiming": false,
                          "_debugNeedsRemount": false,
                          "_debugOwner": null,
                          "_debugSource": null,
                          "actualDuration": 0,
                          "actualStartTime": -1,
                          "alternate": null,
                          "child": null,
                          "childExpirationTime": 0,
                          "dependencies": null,
                          "effectTag": 0,
                          "elementType": null,
                          "expirationTime": 0,
                          "firstEffect": null,
                          "index": 3,
                          "key": null,
                          "lastEffect": null,
                          "memoizedProps": "10",
                          "memoizedState": null,
                          "mode": 0,
                          "nextEffect": null,
                          "pendingProps": "10",
                          "ref": null,
                          "return": [Circular],
                          "selfBaseDuration": 0,
                          "sibling": null,
                          "stateNode": Object {
                            "isHidden": false,
                            "tag": "TEXT",
                            "text": "10",
                          },
                          "tag": 6,
                          "treeBaseDuration": 0,
                          "type": null,
                          "updateQueue": null,
                        },
                        "stateNode": Object {
                          "isHidden": false,
                          "tag": "TEXT",
                          "text": " of ",
                        },
                        "tag": 6,
                        "treeBaseDuration": 0,
                        "type": null,
                        "updateQueue": null,
                      },
                      "stateNode": Object {
                        "isHidden": false,
                        "tag": "TEXT",
                        "text": "5",
                      },
                      "tag": 6,
                      "treeBaseDuration": 0,
                      "type": null,
                      "updateQueue": null,
                    },
                    "stateNode": Object {
                      "isHidden": false,
                      "tag": "TEXT",
                      "text": "Fetched ",
                    },
                    "tag": 6,
                    "treeBaseDuration": 0,
                    "type": null,
                    "updateQueue": null,
                  },
                  "childExpirationTime": 0,
                  "dependencies": null,
                  "effectTag": 0,
                  "elementType": "div",
                  "expirationTime": 0,
                  "firstEffect": null,
                  "index": 0,
                  "key": null,
                  "lastEffect": null,
                  "memoizedProps": Object {
                    "children": Array [
                      "Fetched ",
                      5,
                      " of ",
                      10,
                    ],
                    "style": Object {},
                  },
                  "memoizedState": null,
                  "mode": 0,
                  "nextEffect": null,
                  "pendingProps": Object {
                    "children": Array [
                      "Fetched ",
                      5,
                      " of ",
                      10,
                    ],
                    "style": Object {},
                  },
                  "ref": null,
                  "return": [Circular],
                  "selfBaseDuration": 0,
                  "sibling": null,
                  "stateNode": [Circular],
                  "tag": 5,
                  "treeBaseDuration": 0,
                  "type": "div",
                  "updateQueue": null,
                },
                "isHidden": false,
                "props": Object {
                  "children": Array [
                    "Fetched ",
                    5,
                    " of ",
                    10,
                  ],
                  "style": Object {},
                },
                "rootContainerInstance": [Circular],
                "tag": "INSTANCE",
                "type": "div",
              },
            ],
            "createNodeMock": [Function],
            "tag": "CONTAINER",
          },
          "context": Object {},
          "current": [Circular],
          "finishedExpirationTime": 0,
          "finishedWork": null,
          "firstPendingTime": 0,
          "firstSuspendedTime": 0,
          "hydrate": false,
          "interactionThreadID": 1,
          "lastExpiredTime": 0,
          "lastPingedTime": 0,
          "lastSuspendedTime": 0,
          "memoizedInteractions": Set {},
          "nextKnownPendingLevel": 0,
          "pendingChildren": null,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "tag": 0,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseQueue": Object {
            "callback": null,
            "expirationTime": 1073741823,
            "next": [Circular],
            "payload": Object {
              "element": <_SearchStats
                numFetched={5}
                numFound={10}
                style={Object {}}
              />,
            },
            "priority": 97,
            "suspenseConfig": null,
            "tag": 0,
          },
          "baseState": null,
          "effects": null,
          "shared": Object {
            "pending": null,
          },
        },
      },
      "child": [Circular],
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 0,
      "elementType": null,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <_SearchStats
          numFetched={5}
          numFound={10}
          style={Object {}}
        />,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": FiberRootNode {
        "callbackExpirationTime": 0,
        "callbackNode": null,
        "callbackPriority": 90,
        "containerInfo": Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "isHidden": false,
                  "tag": "TEXT",
                  "text": "Fetched ",
                },
                Object {
                  "isHidden": false,
                  "tag": "TEXT",
                  "text": "5",
                },
                Object {
                  "isHidden": false,
                  "tag": "TEXT",
                  "text": " of ",
                },
                Object {
                  "isHidden": false,
                  "tag": "TEXT",
                  "text": "10",
                },
              ],
              "internalInstanceHandle": FiberNode {
                "_debugHookTypes": null,
                "_debugID": 4,
                "_debugIsCurrentlyTiming": false,
                "_debugNeedsRemount": false,
                "_debugOwner": [Circular],
                "_debugSource": Object {
                  "columnNumber": 3,
                  "fileName": "/Users/lawrencesiden/projects/omdb-film-browser-web/src/components/search-stats.jsx",
                  "lineNumber": 8,
                },
                "actualDuration": 0,
                "actualStartTime": -1,
                "alternate": null,
                "child": FiberNode {
                  "_debugHookTypes": null,
                  "_debugID": 5,
                  "_debugIsCurrentlyTiming": false,
                  "_debugNeedsRemount": false,
                  "_debugOwner": null,
                  "_debugSource": null,
                  "actualDuration": 0,
                  "actualStartTime": -1,
                  "alternate": null,
                  "child": null,
                  "childExpirationTime": 0,
                  "dependencies": null,
                  "effectTag": 0,
                  "elementType": null,
                  "expirationTime": 0,
                  "firstEffect": null,
                  "index": 0,
                  "key": null,
                  "lastEffect": null,
                  "memoizedProps": "Fetched ",
                  "memoizedState": null,
                  "mode": 0,
                  "nextEffect": null,
                  "pendingProps": "Fetched ",
                  "ref": null,
                  "return": [Circular],
                  "selfBaseDuration": 0,
                  "sibling": FiberNode {
                    "_debugHookTypes": null,
                    "_debugID": 6,
                    "_debugIsCurrentlyTiming": false,
                    "_debugNeedsRemount": false,
                    "_debugOwner": null,
                    "_debugSource": null,
                    "actualDuration": 0,
                    "actualStartTime": -1,
                    "alternate": null,
                    "child": null,
                    "childExpirationTime": 0,
                    "dependencies": null,
                    "effectTag": 0,
                    "elementType": null,
                    "expirationTime": 0,
                    "firstEffect": null,
                    "index": 1,
                    "key": null,
                    "lastEffect": null,
                    "memoizedProps": "5",
                    "memoizedState": null,
                    "mode": 0,
                    "nextEffect": null,
                    "pendingProps": "5",
                    "ref": null,
                    "return": [Circular],
                    "selfBaseDuration": 0,
                    "sibling": FiberNode {
                      "_debugHookTypes": null,
                      "_debugID": 7,
                      "_debugIsCurrentlyTiming": false,
                      "_debugNeedsRemount": false,
                      "_debugOwner": null,
                      "_debugSource": null,
                      "actualDuration": 0,
                      "actualStartTime": -1,
                      "alternate": null,
                      "child": null,
                      "childExpirationTime": 0,
                      "dependencies": null,
                      "effectTag": 0,
                      "elementType": null,
                      "expirationTime": 0,
                      "firstEffect": null,
                      "index": 2,
                      "key": null,
                      "lastEffect": null,
                      "memoizedProps": " of ",
                      "memoizedState": null,
                      "mode": 0,
                      "nextEffect": null,
                      "pendingProps": " of ",
                      "ref": null,
                      "return": [Circular],
                      "selfBaseDuration": 0,
                      "sibling": FiberNode {
                        "_debugHookTypes": null,
                        "_debugID": 8,
                        "_debugIsCurrentlyTiming": false,
                        "_debugNeedsRemount": false,
                        "_debugOwner": null,
                        "_debugSource": null,
                        "actualDuration": 0,
                        "actualStartTime": -1,
                        "alternate": null,
                        "child": null,
                        "childExpirationTime": 0,
                        "dependencies": null,
                        "effectTag": 0,
                        "elementType": null,
                        "expirationTime": 0,
                        "firstEffect": null,
                        "index": 3,
                        "key": null,
                        "lastEffect": null,
                        "memoizedProps": "10",
                        "memoizedState": null,
                        "mode": 0,
                        "nextEffect": null,
                        "pendingProps": "10",
                        "ref": null,
                        "return": [Circular],
                        "selfBaseDuration": 0,
                        "sibling": null,
                        "stateNode": Object {
                          "isHidden": false,
                          "tag": "TEXT",
                          "text": "10",
                        },
                        "tag": 6,
                        "treeBaseDuration": 0,
                        "type": null,
                        "updateQueue": null,
                      },
                      "stateNode": Object {
                        "isHidden": false,
                        "tag": "TEXT",
                        "text": " of ",
                      },
                      "tag": 6,
                      "treeBaseDuration": 0,
                      "type": null,
                      "updateQueue": null,
                    },
                    "stateNode": Object {
                      "isHidden": false,
                      "tag": "TEXT",
                      "text": "5",
                    },
                    "tag": 6,
                    "treeBaseDuration": 0,
                    "type": null,
                    "updateQueue": null,
                  },
                  "stateNode": Object {
                    "isHidden": false,
                    "tag": "TEXT",
                    "text": "Fetched ",
                  },
                  "tag": 6,
                  "treeBaseDuration": 0,
                  "type": null,
                  "updateQueue": null,
                },
                "childExpirationTime": 0,
                "dependencies": null,
                "effectTag": 0,
                "elementType": "div",
                "expirationTime": 0,
                "firstEffect": null,
                "index": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": Object {
                  "children": Array [
                    "Fetched ",
                    5,
                    " of ",
                    10,
                  ],
                  "style": Object {},
                },
                "memoizedState": null,
                "mode": 0,
                "nextEffect": null,
                "pendingProps": Object {
                  "children": Array [
                    "Fetched ",
                    5,
                    " of ",
                    10,
                  ],
                  "style": Object {},
                },
                "ref": null,
                "return": [Circular],
                "selfBaseDuration": 0,
                "sibling": null,
                "stateNode": [Circular],
                "tag": 5,
                "treeBaseDuration": 0,
                "type": "div",
                "updateQueue": null,
              },
              "isHidden": false,
              "props": Object {
                "children": Array [
                  "Fetched ",
                  5,
                  " of ",
                  10,
                ],
                "style": Object {},
              },
              "rootContainerInstance": [Circular],
              "tag": "INSTANCE",
              "type": "div",
            },
          ],
          "createNodeMock": [Function],
          "tag": "CONTAINER",
        },
        "context": Object {},
        "current": [Circular],
        "finishedExpirationTime": 0,
        "finishedWork": null,
        "firstPendingTime": 0,
        "firstSuspendedTime": 0,
        "hydrate": false,
        "interactionThreadID": 1,
        "lastExpiredTime": 0,
        "lastPingedTime": 0,
        "lastSuspendedTime": 0,
        "memoizedInteractions": Set {},
        "nextKnownPendingLevel": 0,
        "pendingChildren": null,
        "pendingContext": null,
        "pendingInteractionMap": Map {},
        "pingCache": null,
        "tag": 0,
        "timeoutHandle": -1,
      },
      "tag": 3,
      "treeBaseDuration": 0,
      "type": null,
      "updateQueue": Object {
        "baseQueue": null,
        "baseState": Object {
          "element": <_SearchStats
            numFetched={5}
            numFound={10}
            style={Object {}}
          />,
        },
        "effects": null,
        "shared": Object {
          "pending": null,
        },
      },
    },
    "selfBaseDuration": 0,
    "sibling": null,
    "stateNode": null,
    "tag": 0,
    "treeBaseDuration": 0,
    "type": [Function],
    "updateQueue": null,
  },
}
`;
